src/directions/directions_service.ts

```diff
@@ -267,17 +270,26 @@ export class MigrationDirectionsService {
           start_address: originResponse.formatted_address,
           end_address: destinationResponse.formatted_address,
         });
+
+        if (legGeometry) {
+          // Directions API's overview_path is an array of LatLngs representing the entire course of this route.
+          // Amazon Location provides a Geometry.LineString array for each leg in a route.
+          // To be compatible with Directions API, we will concatenate each leg's Geometry.LineString coordinates
+          routeLineString.push(...legGeometry);
+        }
       });
 
+      const convertedCoords = convertCoordinates(routeLineString);
```

Comment:
Suggestion: It would be easier to avoid lng/lat and lat/lng confusion errors by choosing more explicit names. Ex:
```
const lngLatRouteCoords: number[][] = []; // was routeLineString
...
const latLngRouteCoords = lngLatToLatLngCoords(lngLatRouteCoords);
```

Alternatively, you can put comments above each variable saying what format it is in, but I'm personally more a fan of just putting it in the name.

(The general version of this guideline is "tend towards putting units in variable and function names", like `directionAngleRadians` or `tripTimeSeconds`)